proto/buoyant-cloud-api.proto

```diff
@@ -130,10 +130,71 @@ message CertificateInfo {
   }
 }
 
+
+message Server {
+  bytes server = 1;
+}
+
+message ServerMessage {
+  oneof message {
+    AddServer added = 1;
+    UpdateServer updated = 2;
+    DeleteServer deleted = 3;
+  }
+}
```

Comment:
this add/update/delete pattern fits nicely with the client-go informers, and serves us well with workloads.

on the other hand, it works well for workloads because it's in the context of a `WorkloadStream` API, and also has a periodic `ListWorkloads` to handle reconciliation. if we're using the unary `LinkerdMessage` API, and we miss an add, update, or delete message, we'll be out of sync.

i can think of two options:
1) Replace add/update/delete with a periodic `AuthPolicyInfo` message with all CRDs in one message:
    ```proto
    message AuthPolicyInfo {
      repeated ServerMessage server = 1;
      repeated ServerAuthorizationMessage server_authorization = 2;
    }
    ```
2) Introduce a `AuthPolicyStream` API

i'm thinking option 1 should be sufficient, and it should minimize the increase of surface area between agent and bcloud. option 2 may be overkill. `WorkloadStream` is particularly useful to track changes during rollouts, which we don't care much for with these policy CRDs. open to other options as well.

Reply:
That makes sense and it will be easier to implement for sure (on the agent side). I considered this approach but I had some concerns. First, `ListWorkloads` happens pretty infrequently (10 mins?). Part of the reason for that I think is the fact that we can get slammed with quite a lot of data in the List message, right? The same problem stand for the policy CRDs. Unlike the certificates objects, there could be a lot of them... potentially. So would we go for 10 minutes interval again? In that case it would be pretty non-realtime. If we go for shorter interval, say 1 minute, we risk generating tons of traffic quite frequently and be in a situation where we throw away 100% of the data most of the time, as nothing changes (like we do for certs...). Albeit not so important, me disliking this polling pattern is another reason for going with this approach. WDYT?

Reply:
yeah all good points. `ListWorkloads` is ok every 10 minutes because it's _only_ for reconciliation, we're always up to date because of add/update/delete, unless we miss a message.

i had assumed we'd send policy CRDs once per minute. i also assumed that policy CRDs would be on the order of # of workloads, but are much smaller objects (that also don't include pods)? is this the case, or could we expect many more policy CRDs than workloads? also on default linkerd installs we'd expect zero?